signal/client/client.go

```diff
@@ -101,14 +101,19 @@ func (c *Client) Receive(msgHandler func(msg *proto.Message) error) {
 
 		operation := func() error {
 
-			err := c.connect(c.key.PublicKey().String(), msgHandler)
+			stream, err := c.connect(c.key.PublicKey().String())
 			if err != nil {
 				log.Warnf("disconnected from the Signal Exchange due to an error: %v", err)
 				c.connWg.Add(1)
 				return err
 			}
 
 			backOff.Reset()
+			err = c.receive(stream, msgHandler)
```

Comment:
Same comment as for management client, if we reset things before blocking we risk facing the same issue. We can add backOff.Reset() in the error treatment as well

Reply:
done